### PR TITLE
DoseToWater 

### PR DESCRIPTION
**Pull request by dsarrut**
Hello, 

You will find here a pull request that try to take care of what have been said, being:
- add gamma in the if condition
- consider Electron DEDX if particle is a gamma
- ignore the 'else' condition of the loop
- replace ComputeTotalDEDX with pointer rather dans string comparison

Some remarks:

- when shooting proton, what about all the other generated particles: C13, O16, N14 (to only name a few), they don't directly contribute to the dose to water ? Before the modification, their deposited energy was added via 'proton' DEDX. Now, with this implementation, they are ignored. It seems wrong to me ?

- I implement the ComputeTotalDEDX with pointer, I think it will change anything about speed, but it is clearer, good catch. 

Could you please test and report ? 

Thanks,

### CODE REVIEW

File: source/digits_hits/src/GateDoseActor.cc [hunk 1]
```diff
@@ -379,29 +386,34 @@ void GateDoseActor::UserSteppingActionInVoxel(const int index, const G4Step* ste
...
-        //if (PartName != "O16[0.0]" && PartName != "alpha" && PartName != "Be7[0.0]" && PartName != "C12[0.0]"){
-
-        DEDX = emcalc->ComputeTotalDEDX(Energy, PartName, material, cut);
-        DEDX_Water = emcalc->ComputeTotalDEDX(Energy, PartName, "G4_WATER", cut);
-        doseToWater=edep/density/Volume/gray*(DEDX_Water/1.)/(DEDX/(density*e_SI));
-        if (DEDX_Water == 0 || DEDX == 0) doseToWater = 0.0; // to avoid inf or NaN
+      double volume = mDoseToWaterImage.GetVoxelVolume();
+
+      // Get current particle
+      const G4ParticleDefinition * p = step->GetTrack()->GetParticleDefinition();
```
Comment (erialm) [inline source/digits_hits/src/GateDoseActor.cc]: Get rid of the const here, I think. Line 406 will cause problems trying to assign a new value to a const.

Comment (dsarrut) [inline source/digits_hits/src/GateDoseActor.cc]: Not sure ... I think it is a pointer to const and not a const pointer, meaning the object pointed to is const.

Comment (djboersma) [inline source/digits_hits/src/GateDoseActor.cc]: I agree with @dsarrut here. :)

**Comment by GitFuchs** [comment]
Is there a reason why DEDX is not calculated for every particle (performance etc) ?
Like: If not a photon do...

**Pull request merged**